### PR TITLE
E2e kubeadm new test

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
This PR adds new E2E kubeadm tests

**Special notes for your reviewer**:
This PR is built on top of https://github.com/kubernetes/kubernetes/pull/75640, please check only the last commit

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/sig cluster-lifecycle
/area kubeadm
/priority backlog

@kubernetes/sig-cluster-lifecycle-pr-reviews
/assign @neolit123 @rosti
/cc @ereslibre
@MalloZup

### CODE REVIEW

File: test/e2e_kubeadm/dns_addon_test.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+	kubeDNSDeploymentName     = "kube-dns"
+)
+
+var (
+	dnsType = ""
+)
+
+// Define container for all the test specification aimed at verifying
+// that kubeadm configures the dns as expected
+var _ = KubeadmDescribe("dns addon", func() {
```
Comment (neolit123) [inline test/e2e_kubeadm/dns_addon_test.go]: dns -> DNS
does it matter if it's lowercase only?

File: test/e2e_kubeadm/dns_addon_test.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+	// Get an instance of the k8s test framework
+	f := framework.NewDefaultFramework("dns")
```
Comment (neolit123) [inline test/e2e_kubeadm/dns_addon_test.go]: dns -> DNS
does it matter if it's lowercase only?

File: test/e2e_kubeadm/dns_addon_test.go [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+				}
+
+				d := GetDeployment(f.ClientSet, kubeSystemNamespace, kubeDNSDeploymentName)
+
+				Expect(d.Spec.Template.Spec.ServiceAccountName).To(Equal(kubeDNSServiceAccountName))
+			})
+		})
+	})
+
+	Context("coreDNS", func() {
```
Comment (neolit123) [inline test/e2e_kubeadm/dns_addon_test.go]: `CoreDNS`

File: test/e2e_kubeadm/dns_addon_test.go [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+				}
+
+				ExpectServiceAccount(f.ClientSet, kubeSystemNamespace, kubeDNSServiceAccountName)
+			})
+		})
+
+		Context("kube-dns Deployment", func() {
+			It("should exist and be properly configured", func() {
+				if dnsType != "kube-dns" {
+					framework.Skipf("Skipping because proxy type is %s", dnsType)
```
Comment (neolit123) [inline test/e2e_kubeadm/dns_addon_test.go]: `proxy type` - > `DNS type`

File: test/e2e_kubeadm/dns_addon_test.go [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+				Expect(d.Spec.Template.Spec.ServiceAccountName).To(Equal(kubeDNSServiceAccountName))
+			})
+		})
+	})
+
+	Context("coreDNS", func() {
+		Context("coreDNS ServiceAccount", func() {
+			It("should exist", func() {
+				if dnsType != "CoreDNS" {
+					framework.Skipf("Skipping because proxy type is %s", dnsType)
```
Comment (neolit123) [inline test/e2e_kubeadm/dns_addon_test.go]: `proxy type - > DNS type`

File: test/e2e_kubeadm/dns_addon_test.go [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+
+				d := GetDeployment(f.ClientSet, kubeSystemNamespace, kubeDNSDeploymentName)
+
+				Expect(d.Spec.Template.Spec.ServiceAccountName).To(Equal(kubeDNSServiceAccountName))
+			})
+		})
+	})
+
+	Context("coreDNS", func() {
+		Context("coreDNS ServiceAccount", func() {
```
Comment (neolit123) [inline test/e2e_kubeadm/dns_addon_test.go]: `CoreDNS`

File: test/e2e_kubeadm/dns_addon_test.go [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+				if dnsType != "CoreDNS" {
+					framework.Skipf("Skipping because proxy type is %s", dnsType)
+				}
+
+				ExpectServiceAccount(f.ClientSet, kubeSystemNamespace, coreDNSServiceAccountName)
+			})
+
+			It("should have related ClusterRole and ClusterRoleBinding", func() {
+				if dnsType != "CoreDNS" {
+					framework.Skipf("Skipping because proxy type is %s", dnsType)
```
Comment (neolit123) [inline test/e2e_kubeadm/dns_addon_test.go]: proxy type - > DNS type

File: test/e2e_kubeadm/dns_addon_test.go [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+				}
+
+				ExpectClusterRole(f.ClientSet, coreDNSRoleName)
+				ExpectClusterRoleBinding(f.ClientSet, coreDNSRoleBindingName)
+			})
+		})
+
+		Context("coreDNS ConfigMap", func() {
```
Comment (neolit123) [inline test/e2e_kubeadm/dns_addon_test.go]: CoreDNS

File: test/e2e_kubeadm/dns_addon_test.go [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+					framework.Skipf("Skipping because proxy type is %s", dnsType)
+				}
+
+				cm := GetConfigMap(f.ClientSet, kubeSystemNamespace, coreDNSConfigMap)
+
+				Expect(cm.Data).To(HaveKey(coreDNSConfigMapKey))
+			})
+		})
+
+		Context("coreDNS Deployment", func() {
```
Comment (neolit123) [inline test/e2e_kubeadm/dns_addon_test.go]: CoreDNS

File: test/e2e_kubeadm/dns_addon_test.go [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+
+				ExpectClusterRole(f.ClientSet, coreDNSRoleName)
+				ExpectClusterRoleBinding(f.ClientSet, coreDNSRoleBindingName)
+			})
+		})
+
+		Context("coreDNS ConfigMap", func() {
+			It("should exist and be properly configured", func() {
+				if dnsType != "CoreDNS" {
+					framework.Skipf("Skipping because proxy type is %s", dnsType)
```
Comment (neolit123) [inline test/e2e_kubeadm/dns_addon_test.go]: proxy type - > DNS type

File: test/e2e_kubeadm/dns_addon_test.go [hunk 11]
```diff
@@ -0,0 +1,150 @@
...
+				cm := GetConfigMap(f.ClientSet, kubeSystemNamespace, coreDNSConfigMap)
+
+				Expect(cm.Data).To(HaveKey(coreDNSConfigMapKey))
+			})
+		})
+
+		Context("coreDNS Deployment", func() {
+			It("should exist and be properly configured", func() {
+				if dnsType != "CoreDNS" {
+					framework.Skipf("Skipping because proxy type is %s", dnsType)
```
Comment (neolit123) [inline test/e2e_kubeadm/dns_addon_test.go]: proxy type - > DNS type
should we add a wrapper / helper that has the message only one time and does the framework skip?
e.g. `maybeSkipDNSType(framework, dnsType, "CoreDNS")`

File: test/e2e_kubeadm/kubeadm_certs_test.go [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+
+	// Tests in this container are not expected to create new objects in the cluster
+	// so we are disabling the creation of a namespace in order to get a faster execution
+	f.SkipNamespaceCreation = true
+
+	It("should exist and be properly configured", func() {
+		s := GetSecret(f.ClientSet, kubeSystemNamespace, kubeadmCertsSecretName)
+
+		// Checks the kubeadm-certs is ownen by a time lived token
+		Expect(s.OwnerReferences).To(HaveLen(1), "%s should have an owner reference", kubeadmCertsSecretName)
```
Comment (neolit123) [inline test/e2e_kubeadm/kubeadm_certs_test.go]: possibly
`%s should have one owner reference`

File: test/e2e_kubeadm/kubelet_config_test.go [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+	})
+
+	It("should be accessible for bootstrap tokens", func() {
+		ExpectSubjectHasAccessToResource(f.ClientSet,
+			rbacv1.GroupKind, bootstrapTokensGroup,
+			kubeadmConfigConfigMapResource,
+		)
+	})
+
+	It("should be accessible for for nodes", func() {
```
Comment (neolit123) [inline test/e2e_kubeadm/kubelet_config_test.go]: double `for`

File: test/e2e_kubeadm/nodes_test.go [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+var _ = KubeadmDescribe("nodes", func() {
+
+	// Get an instance of the k8s test framework
+	f := framework.NewDefaultFramework("nodes")
+
+	// Tests in this container are not expected to create new objects in the cluster
+	// so we are disabling the creation of a namespace in order to get a faster execution
+	f.SkipNamespaceCreation = true
+
+	It("should have cri annotation", func() {
```
Comment (neolit123) [inline test/e2e_kubeadm/nodes_test.go]: cri -> CRI

**Review by neolit123 — COMMENTED** [review]
thanks @fabriziopandini ! :tada: 
did a first pass and this SGTM, added minor comments about naming.

/assign @timothysc

**Review by dixudx — APPROVED** [review]
/lgtm

**Pull request merged**